tests/filecheck/dialects/smt/ops.mlir

```diff
@@ -10,6 +10,11 @@
 %func1 = smt.declare_fun : !smt.func<() !smt.bool>
 %func2 = smt.declare_fun : !smt.func<(!smt.bool) !smt.bool>
 
+// CHECK-NEXT:    smt.apply_func %func1() : !smt.func<() !smt.bool>
+// CHECK-NEXT:    smt.apply_func %func2(%const1) : !smt.func<(!smt.bool) !smt.bool>
+smt.apply_func %func1() : !smt.func<() !smt.bool>
+smt.apply_func %func2(%const1) : !smt.func<(!smt.bool) !smt.bool>
```

Comment:
Worth including the return value here?

Reply:
I'm copying MLIR syntax here (I'll add the compatible tests once we have a version that includes SMT).
The return type is the second `smt.bool` in the `smt.func`